addons/stock/wizard/make_procurement_product.py

```diff
@@ -2,106 +2,91 @@
-        user = self.pool.get('res.users').browse(cr, uid, uid, context=context).login
-        wh_obj = self.pool.get('stock.warehouse')
-        procurement_obj = self.pool.get('procurement.order')
-        data_obj = self.pool.get('ir.model.data')
-
-        for proc in self.browse(cr, uid, ids, context=context):
-            wh = wh_obj.browse(cr, uid, proc.warehouse_id.id, context=context)
-            procure_id = procurement_obj.create(cr, uid, {
-                'name':'INT: '+str(user),
+        ProcurementOrder = self.env['procurement.order']
+
+        for proc in self:
+            procurement = ProcurementOrder.create({
+                'name': 'INT: '+str(self.env.user.login),
                 'date_planned': proc.date_planned,
                 'product_id': proc.product_id.id,
                 'product_qty': proc.qty,
                 'product_uom': proc.uom_id.id,
                 'warehouse_id': proc.warehouse_id.id,
-                'location_id': wh.lot_stock_id.id,
-                'company_id': wh.company_id.id,
+                'location_id': proc.warehouse_id.lot_stock_id.id,
+                'company_id': proc.warehouse_id.company_id.id,
                 'route_ids': [(6, 0, proc.route_ids.ids)],
             })
-            procurement_obj.signal_workflow(cr, uid, [procure_id], 'button_confirm')
-
-        id2 = data_obj._get_id(cr, uid, 'procurement', 'procurement_tree_view')
-        id3 = data_obj._get_id(cr, uid, 'procurement', 'procurement_form_view')
+            procurement.signal_workflow('button_confirm')
 
-        if id2:
-            id2 = data_obj.browse(cr, uid, id2, context=context).res_id
-        if id3:
-            id3 = data_obj.browse(cr, uid, id3, context=context).res_id
+        proc_tree = self.env.ref('procurement.procurement_tree_view')
+        proc_form = self.env.ref('procurement.procurement_form_view')
 
         return {
             'view_type': 'form',
             'view_mode': 'tree,form',
             'res_model': 'procurement.order',
-            'res_id' : procure_id,
-            'views': [(id3,'form'),(id2,'tree')],
+            'res_id': procurement.id,
+            'views': [(proc_form.id, 'form'), (proc_tree.id, 'tree')],
             'type': 'ir.actions.act_window',
-         }
+        }
 
-    def default_get(self, cr, uid, fields, context=None):
-        if context is None:
-            context = {}
-        record_id = context.get('active_id')
+    @api.model
+    def default_get(self, fields):
+        record_id = self._context.get('active_id')
 
-        if context.get('active_model') == 'product.template':
-            product_ids = self.pool.get('product.product').search(cr, uid, [('product_tmpl_id', '=', context.get('active_id'))], context=context)
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.product'].search([('product_tmpl_id', '=', self._context.get('active_id'))])
```

Comment:
1.variable name is represent its values.
2.search returns recordset not ids so varibale name should be 
```
> products
> products_rec
```